zerver/lib/reminders.py

```diff
@@ -12,24 +14,47 @@
 from zerver.models.scheduled_jobs import ScheduledMessage
 
 
-def get_reminder_formatted_content(message: Message, current_user: UserProfile) -> str:
+def get_reminder_formatted_content(
+    message: Message, current_user: UserProfile, note_text: str = ""
+) -> str:
+    note_text = re.sub(r"\n+", " ", note_text.strip())
     if message.is_stream_message():
         # We don't need to check access here since we already have the message
         # whose access has already been checked by the caller.
         stream = Stream.objects.get(
             id=message.recipient.type_id,
             realm=current_user.realm,
         )
-        content = _("You requested a reminder for {message_pretty_link}.").format(
-            message_pretty_link=get_message_link_syntax(
-                stream_id=stream.id,
-                stream_name=stream.name,
-                topic_name=message.topic_name(),
-                message_id=message.id,
+        if note_text:
+            content = _(
+                "You requested a reminder for {message_pretty_link}. Note:\n > {note_text}"
+            ).format(
+                message_pretty_link=get_message_link_syntax(
+                    stream_id=stream.id,
+                    stream_name=stream.name,
+                    topic_name=message.topic_name(),
+                    message_id=message.id,
+                ),
```

Comment:
Let's store `message_pretty_link` before the `if` block to avoid code repetition.